File: programmieren lernen/python/quelldateien/03_wuerfel_kommentiert.py
# Ein Programm, das so oft eine Zahl würfelt, bis der Spieler "stopp" eintippt

# das Modul "random" wird eingebettet,
# damit wir später eine Zufallszahl erzeugen können
import random

# eine Schleife - alle eingerückten Zeilen werden immer wieder wiederholt
while True:
    # eine Zufallszahl zwischen 1 und 6 wird in der Variablen "zahl" gespeichert
    zahl = random.randint(1, 6)
    # ...und in eine Zeichenkette umgewandelt (damit man sie später ausgeben kann)
    zahl = str(zahl)

    # die "gewürfelte" Zahl ausgeben
    print("Ich habe eine " + zahl + " gewürfelt.")

    # dem Benutzer die Chance geben, etwas einzutippen (oder ENTER zu drücken)
    befehl = input()

    # wenn der Benutzer "stopp" eingegeben hat...
    if befehl == "stopp":
        print("Schade, schon vorbei.")
        # ...die Schleife beenden
        break

File: programmieren lernen/cpp/quelldateien/03_wuerfel_kommentiert.cpp
// Ein Programm, das so oft eine Zahl würfelt, bis der Spieler "stopp" eintippt

#include <iostream>
#include <stdlib.h>

using namespace std;

int main(void){ 
	
	srand(time(NULL));	// Wird für die Zufallszahlenerzeugung benötigt
	int zufall;	// Die gewürfelte Zahl
	string schleife;	// Die Eingabe des Spielers

	// Eine Schleife - Alles bis 'while' wird wiederholt
	do{
		// Eine Zufallszahl zwischen 1 und 6 wird in der Variablen "zufall" gespeichert
		zufall = rand() % 6 + 1;

		// Die "gewürfelte" Zahl ausgeben
		cout << endl << "Ich habe eine " << zufall << " gewuerfelt!" << endl;
		
		// Der Benutzer darf hier etwas eingeben
		cin >> schleife;

	// War die Eingabe nicht "stopp", wird die Schleife wiederholt
	}while(schleife != "stopp");

	cout << endl << "Schade, schon vorbei." << endl;

	// Das Programm beenden
	return 0;
}

